Modules/Module_Import/Import/xASL_imp_DetermineSubjectStructure.m

```diff
@@ -113,6 +113,7 @@
     
 	% Initialize an empty list of visits
 	listVisits = {};
+	bVisitsDetected = false; % Set to true once at least a single visit is detected
```

Comment:
We need to be sparingly with adding more booleans, I would only add new booleans if really needed, because the more variables the more difficult code becomes to read.

Probably a nicer way to do this, is replace bVisitsDetected throughout by numel(listSubjectsVisits)>0 [or ~isempty(listSubjectsVisits)]

-> smaller change in github history
-> less booleans -> easier to read (you don't have to search back in the code what the boolean means)

ALSO: note that now `bVisitsDetected` has no meaning for the use case where tokenVisitAliases was defined. This makes it confusing to read, because what happens when `bVisitsDetected` is false but tokenVisitAliases was defined? This should give a warning or whatsoever. You would answer me that this doesn't matter, is taken into account in other parts of the code. And then the answer of the first-time reader is "but then the general `bVisitsDetected` is a confusing boolean. -> Hence, should be replaced by a non-boolean (or call the boolean bVisitsDetected_UseOnlyWhenNoTokenVisitAliasesAreDefined which is senseless ;)

Reply:
I made more comments in the code. But we can't replace the boolean in the way you suggested. Because that would simply not work at all.

Your suggestion "numel(listSubjectsVisits)>0" wouldn't work as this list contains a mixture of subject-visits. We are not interested in subject-visits but in what visits we have. We are primarily interested if the subjects are "sub1_" - where visit name is missing. Or "sub1" where visit name wasn't defined. It is really important to know if visit wasn't defined so we ignore missing visits. Or in case where visit names are simply missing and we need to replace this with "missingSession value".

However, we are constructing listVisits only to recreate tokenVisitAliases if needed. So we can skip nonexisting visits and only list empty visits. 

So i removed the boolean and changed the way the listVisit is indexed.
Done. You can resolve this.

Reply:
Thanks.
Sorry I now saw only the changes, with the residual code compressed in between. So it seemed as if 

bVisitsDetected == numel(listSubjectsVisits)>0

but I now see this is not the case.

So you can ignore this comment, and resolve once you read this.

Reply:
Yes. I know. It looked simple and the initial implementation was like this, until I started testing properly :)